driver-core/src/test/unit/com/mongodb/internal/async/AsyncFunctionsAbstractTest.java

```diff
@@ -480,6 +505,29 @@ void testTryCatch() {
                 });
     }
 
+    @Test
+    void testTryWithEmptyCatch() {
+        assertBehavesSameVariations(4,
+                () -> {
+                    try {
+                        sync(1);
+                    } catch (MongoException e) {
+                        // ignore exceptions
+                    } finally {
+                        plain(2);
+                    }
+                },
+                (callback) -> {
+                    beginAsync().thenRun(c -> {
+                        async(1, c);
+                    })
+                    // ignore exceptions when releasing server resources
```

Comment:
Actually, I have doubts it's the same as the sync code, because the sync code ignores only `MongoException`s.